base/ltsockets.dtx

```diff
@@ -472,7 +472,7 @@
 %   Socket plug names have to be unique within on a per socket basis,
 %   but it is perfectly allowed (and sensible in some cases) to use
 %   the same plug name with different sockets (where based on the
-%   socket different actions may be associated with the plug
+%   sockets different actions may be associated with the plug
 %   name). For example \plug{noop} is a plug name declared for every
```

Comment:
@car222222 It seems to me here "different actions" is the verb, hence "based on the sockets" is sth like a conditional adverbial. Maybe "where based on the sockets' purposes, different actions may be associated with the plug name" is clearer?

Reply:
I need to look at this whole section, including the concept itself.

I hope to be able to do that later this week.